kit/src/main/java/com/oracle/javafx/scenebuilder/kit/fxom/FXOMLoader.java

```diff
@@ -95,14 +103,31 @@ public void load(String fxmlText) throws java.io.IOException {
             is.reset();
             setSceneGraphRoot(fxmlLoader.load(is));
         } catch (RuntimeException | IOException x) {
-            if (x.getCause().getClass() == XMLStreamException.class) {
-                handleUnsupportedCharset(x);
-            } else
-                throw new IOException(x);
+            handleFxmlLoadingError(x);
+        }
+    }
+
+    private void handleFxmlLoadingError(Exception x) throws IOException {
+        if (x.getCause() != null) {
+            handleKnownCauses(x);
+        } else {
+            handleUnknownAndMissingCauses(x);
+        }
+    }
+
+    private void handleUnknownAndMissingCauses(Exception x) throws IOException {
+        throw new IOException(x);
```

Comment:
I assume throwing this is fine as it will be caught by the SB general exception handler?

Reply:
I do not actually know why there was a separation between the path where causes are shown and where just an IOException is raised. The branching was already there. Technically, I think, that the branching is most likely not needed and we just could use the exception handler here to show the error. 

IOException and Runtime exceptions will eventually cause an IllegalStateException in FXOMRefresher and the FXML causing the problem will be dumped. There is no proper error handling yet which becomes visible in GUI.

As of now, I would not change this as this exceeds the scope of the PR and I cannot oversee consequences yet. For further changes here, I'd like to add more tests first.

Reply:
ok